s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -231,6 +237,36 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
     }
   }
 
+  def multipartCopy(
+      sourceLocation: S3Location,
+      targetLocation: S3Location,
+      sourceVersionId: Option[String] = None,
+      contentType: ContentType = ContentTypes.`application/octet-stream`,
+      s3Headers: S3Headers,
+      sse: Option[ServerSideEncryption] = None,
+      chunkSize: Int = MinChunkSize,
+      chunkingParallelism: Int = 4
+  ): RunnableGraph[Future[CompleteMultipartUploadResult]] = {
+    import mat.executionContext
+
+    // Pre step get source meta to get content length (size of the object)
+    val eventualMaybeObjectSize: Future[Option[Long]] =
+      getObjectMetadata(sourceLocation.bucket, sourceLocation.key, sse).map(_.map(_.contentLength))
+    val eventualPartitions =
+      eventualMaybeObjectSize.map(_.map(createPartitions(chunkSize, sourceLocation)).getOrElse(Nil))
+    val partitions = Source.fromFuture(eventualPartitions)
```

Comment:
Why do you wrap it into a stream? Isn't the future good enough?

Reply:
First step in multi part copy (or upload) is to initiate multi part upload, which is done in `initiateUpload` (which is an existing function, used in `multipartUpload`, the return type of this function is `Source[(MultipartUpload, Int), NotUsed]`) , in order to copy an object first step is to get the content length of the existing object which is done here, the reason for converting into Source is to zip with source of `initiateUpload`.